### PR TITLE
#167165082 Implements Get  Comment Edit History

### PR DESCRIPTION
**Pull request by OdunayoOkebunmi**
### What does this PR do?
- user can see previous versions of comments
- adds tests to cover edge cases
- add swagger documentation
#### Description of Task to be completed?
As a user, I should be able to see the previous versions of comments made on an article after previous edits.
API Flowchart (7)

#### How should this be manually tested?
After cloning this repo, `cd` into it and do the following:

```bash
# fetch this branch
$ git fetch origin  ft-comment-edit-history-167165082

# install dependencies
$ npm install

# run unit tests
$ npm test

# start the development server
$ npm run server
```
Using postman, run /api/v1/admin/comments/{id} 

#### What are the relevant pivotal tracker stories?
[#1167165082](https://www.pivotaltracker.com/story/show/167165082)

#### Screenshots (if appropriate)
Screenshot 2019-08-15 at 12 01 02

### CODE REVIEW

File: controllers/CommentController.js [hunk 1]
```diff
@@ -137,4 +138,78 @@ export default class CommentController {
...
+        where: { commentId },
+        order: [
+          ['updatedAt', 'DESC'],
+        ],
+      });
+      if (!commentHistory[0]) {
+        return errorResponse(res, 404, { message: 'Comment does not have edit history' });
+      }
+      if (commentHistory[0].dataValues.userId !== id) {
+        return errorResponse(res, 403, 'You are not allowed to edit another user\'s comment');
```
Comment (PelumiAlesh) [inline controllers/CommentController.js]: @OdunayoOkebunmi  Is this error message correct? Is this not meant to be 'You are not allowed to view another user\'s comment history'

**Review by profmcdan — CHANGES_REQUESTED** [review]
Your present implementation is not particularly correct, this is meant to be an admin feature, so its the admin or anyone with the required privileges that should be able to pull edit history on a comment.

Try to attend to this.

File: controllers/CommentController.js [hunk 2]
```diff
@@ -174,4 +180,36 @@ export default class CommentController {
...
+        where: { commentId },
+        order: [
+          ['updatedAt', 'DESC'],
+        ],
+      });
+      if (req.user.type === 'user') {
+        return errorResponse(res, 403, { message: 'You are not authorised to view comment history' });
+      }
+      if (!commentHistory[0]) {
+        return errorResponse(res, 404, { message: 'Comment does not have edit history' });
```
Comment (tolulope-od) [inline controllers/CommentController.js]: Wouldn't it be better to just return the empty array here?

**Review by profmcdan — APPROVED** [review]
